pkg/utils/kubernetes/utils.go

```diff
@@ -4,31 +4,30 @@ import (
 	"errors"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/discovery"
-	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
-func CustomResourceDefinitionExists(gvk schema.GroupVersionKind) (bool, error) {
+func CustomResourceDefinitionExists(gvk schema.GroupVersionKind) error {
 	cfg, err := config.GetConfig()
 	if err != nil {
-		return false, err
+		return err
 	}
-	return customResourceDefinitionExists(gvk, cfg)
-}
-
-func customResourceDefinitionExists(gvk schema.GroupVersionKind, cfg *rest.Config) (bool, error) {
 	client, err := discovery.NewDiscoveryClientForConfig(cfg)
 	if err != nil {
-		return false, err
+		return err
 	}
+	return customResourceDefinitionExists(gvk, client)
+}
+
+func customResourceDefinitionExists(gvk schema.GroupVersionKind, client discovery.ServerResourcesInterface) error {
```

Comment:
I see, mainly to change the code to make certain part of the code more testable. So a code's testability out weight code's entirety?
What problem will cause if we use a fake web server to test the previous func as a whole?

Reply:
Yes, split them to allow providing client and to unit test.